test/message-input.test.js

```diff
@@ -58,5 +58,35 @@ describe('message-input', () => {
       keyDownOn(textArea.inputElement, 13, ['shift'], 'Enter');
       expect(spy.called).to.be.false;
     });
+
+    it('should not fire a submit event if value is empty', () => {
+      const spy = sinon.spy();
+      messageInput.addEventListener('submit', spy);
+      button.click();
+      expect(spy.notCalled).to.be.true;
+    });
+
+    it('should empty input after submit', () => {
+      messageInput.value = 'foo';
+      expect(messageInput.value).to.be.equal('foo');
+      button.click();
+      expect(messageInput.value).to.be.empty;
+    });
+
+    it('should contain filled value in submit event', (done) => {
+      messageInput.addEventListener('submit', (e) => {
```

Comment:
```suggestion
      const spy = sinon.spy();
      messageInput.addEventListener('submit', spy);
      textArea.value = 'foo';
      button.click();
      const e = spy.firstCall.args[0];
      expect(e.detail.value).to.be.equal('foo')
```

Let's use the spy `args` to get the event, this way we don't need to make the test async.